lib/measures/tbd/measure.rb

```diff
@@ -50,16 +66,33 @@ def run(model, runner, user_arguments)
     super(model, runner, user_arguments)
 
     # assign the user inputs to variables
+    load_tbd_json = runner.getBoolArgumentValue("load_tbd_json", user_arguments)
+
     option = runner.getStringArgumentValue("option", user_arguments)
-    psi = PSI.new
-    set = psi.set[option]
+
+    write_tbd_json = runner.getBoolArgumentValue("write_tbd_json", user_arguments)
 
     # use the built-in error checking
     if !runner.validateUserArguments(arguments(model), user_arguments)
       return false
     end
 
-    surfaces = processTBD(model, set)
+    io_path = nil
+    schema_path = nil
+
+    if load_tbd_json
+      io_path = runner.workflow.findFile('tbd.json')
+      if io_path.empty?
+        runner.registerError("Cannot find tbd.json")
+        return false
+      else
+        io_path = io_path.get.to_s
+        runner.registerInfo("Using inputs from #{io_path}")
+      end
+    end
+
+    # DLM: can processTBD also return the content of the TBD JSON to write?
+    surfaces = processTBD(model, option, io_path, schema_path)
```

Comment:
@rd2 I think it would be good to return the TDB JSON struct to write from `processTBD` in addition to `surfaces`, is that possible?

Reply:
Sure. I'll start working on this.